src/controllers/auth.controller.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const { User } = require('../models/user.js');
+const emailService = require('../services/email.service.js');
+const { v4: uuidv4 } = require('uuid');
+
+const register = async(req, res) => {
+  const { name, email, password } = req.body;
+  const activationToken = uuidv4();
+
+  const newUser = await User.create({
+    name,
+    email,
+    password,
+    activationToken,
+  });
+
+  emailService.sendActivationEmail(email, activationToken);
+  res.send(newUser);
+};
```

Comment:
to much logic for controller. it should be responsible for taking data from request ant call service method with it, for ex:
```js
const register = async(req, res) => {
  try {
    const user = userService.create(req.body);
    res.send(user);
  } catch (error) {
    console.log(error.message);
    res.send(400);
  }
};
```

Reply:
please fix it everywhere